packages/agreement-process/src/services/agreementSuspensionProcessor.ts

```diff
@@ -111,39 +68,15 @@ export async function suspendAgreementLogic({
     suspendedByConsumer,
     suspendedByProducer,
     stamps: {
-      ...agreement.data.stamps,
+      ...agreement.stamps,
       suspensionByConsumer: suspensionByConsumerStamp,
       suspensionByProducer: suspensionByProducerStamp,
     },
-    suspendedAt: agreement.data.suspendedAt ?? new Date(),
+    suspendedAt: agreement.suspendedAt ?? new Date(),
   };
 
-  const updatedAgreement: Agreement = {
-    ...agreement.data,
+  return {
+    ...agreement,
     ...updateSeed,
   };
-
-  if (authData.organizationId === agreement.data.producerId) {
-    return [
-      updatedAgreement,
-      toCreateEventAgreementSuspendedByProducer(
-        updatedAgreement,
-        agreement.metadata.version,
-        correlationId
-      ),
-    ];
-  } else if (authData.organizationId === agreement.data.consumerId) {
-    return [
-      updatedAgreement,
-      toCreateEventAgreementSuspendedByConsumer(
-        updatedAgreement,
-        agreement.metadata.version,
-        correlationId
-      ),
-    ];
-  } else {
-    throw new Error(
-      "Unexpected organizationId: Agreement can be suspended only by consumer or producer"
-    );
-  }
```

Comment:
Why not keeping the event creation here as you do in the other processor methods?